plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/BuildVersionMetaDataPublisherTest.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.grandhackathon.deployment.sphere.plugin;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.isA;
+
+import java.io.PrintStream;
+
+import hudson.model.Action;
+import hudson.model.BuildListener;
+import hudson.model.FreeStyleBuild;
+import hudson.model.AbstractBuild;
+import hudson.tasks.BuildStepMonitor;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.epam.grandhackathon.deployment.sphere.plugin.action.DynamicVariablesStorageAction;
+
+public class BuildVersionMetaDataPublisherTest {
+
+	@Test
+	public void shouldReturnMonitorServiceBuild() throws Exception {
+		BuildVersionMetaDataPublisher publisher = new BuildVersionMetaDataPublisher();
+		assertEquals("Illegal required monitor service ", publisher.getRequiredMonitorService(), BuildStepMonitor.BUILD);
+	}
+
+	@Test
+	public void shouldOnPerformRejectNullValues() throws Exception {
```

Comment:
Based on the comments:
1. I should rename method to shouldThrownOnPerformWithNullBuild
2. Instead of  Anotation @Test(expected = IllegalArgumentException.class) i should use following code
```
    @Rule
    public ExpectedException thrown = ExpectedException.none();
    // .... 

    @Test
	public void shouldThrownOnPerformWithNullBuild() throws Exception {
        thrown.expect(IllegalArgumentException.class);
        thrown.expectMessage("Build value is null. Build value must be provided");
        BuildVersionMetaDataPublisher publisher = new BuildVersionMetaDataPublisher();
		BuildListener buildListener = mock(BuildListener.class);
        publisher.perform((AbstractBuild)null, null, buildListener);
	}
```

smth else ?